types/state/stateTxn.go

```diff
@@ -0,0 +1,334 @@
+package state
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"github.com/AccumulateNetwork/accumulate/internal/logging"
+	"github.com/AccumulateNetwork/accumulate/smt/common"
+	"github.com/AccumulateNetwork/accumulate/smt/managed"
+	"github.com/AccumulateNetwork/accumulate/smt/storage"
+	"github.com/AccumulateNetwork/accumulate/smt/storage/database"
+	"github.com/AccumulateNetwork/accumulate/types"
+	"sort"
+	"sync"
+)
+
+type DBTransaction struct {
+	state *StateDB
```

Comment:
`updates`, `writes`, and `transactions` need to be removed from `StateDB` and added to `DBTransaction`.